srt-tokio/src/listener/mod.rs

```diff
@@ -0,0 +1,150 @@
+mod request;
+mod state;
+
+pub use request::ConnectionRequest;
+
+use std::io;
+
+use futures::{channel::mpsc, prelude::*};
+use srt_protocol::settings::ConnInitSettings;
+use tokio::{net::ToSocketAddrs, task::JoinHandle};
+
+use crate::watch;
+
+use super::net::PacketSocket;
+
+pub use srt_protocol::listener::ListenerStatistics;
+
+pub struct SrtListener {
+    settings: ConnInitSettings,
+    request_receiver: mpsc::Receiver<ConnectionRequest>,
+    statistics_receiver: watch::Receiver<ListenerStatistics>,
+    _task: JoinHandle<()>,
```

Comment:
What's the reason for holding onto this? Afaik JoinHandles don't have any special `Drop` behavior

Reply:
I was pondering how to handle Listener shutdown. Should we join on all the open socket handles? Again, this is a rough draft.